File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        arrWord = []
        newWord = ''

        if len(word1) > len(word2):
            j = len(word2)
        else:
            j = len(word1)
        i = 0

        for i in range(j):
            arrWord.append(word1[i])
            arrWord.append(word2[i])
        
        arrWord += word1[j:] if len(word1) > len(word2) else word2[j:]
        newWord = newWord.join(arrWord)
            
        return newWord

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.cpp
// .cpp solution for leetcode
class Solution {
public:
    string mergeAlternately(string word1, string word2) {

    string newWord = "";
    
    int j = min(word1.length(), word2.length());

    for (int i = 0; i < j; i++){
        newWord += word1[i];
        newWord += word2[i];
    }

    newWord += (word1.length() > word2.length()) ? word1.substr(j) : word2.substr(j);
    
    return newWord;
    }
};
